NetCoreMentoring/src/NetCoreMentoring.Core/Services/ProductService.cs

```diff
@@ -37,11 +37,11 @@ public Product GetProduct(int id)
 
         public IEnumerable<Product> GetProducts()
         {
-            //TODO: add exception handling
+            // TODO: add error handling
             var maxProductsOnPage = int.Parse(_configuration["MaxProductsOnPage"]);
 
             var result = _context.Products
-                .Take(maxProductsOnPage)
+                .Take(maxProductsOnPage == 0 ? int.MaxValue : maxProductsOnPage)
```

Comment:
I think this is a wrong design, in any case the query that doesn't contain `where/limit statement` will be more effective than the query with this condition. I know maybe some `db`s can optimize it via indexes or so, but in any case this is wrong from the design point of view. The `==0` code path should not contain any filtering conditions at all.

Reply:
done